archive/tar.go

```diff
@@ -393,7 +393,8 @@ func createTarFile(ctx context.Context, path, extractDir string, hdr *tar.Header
 		if strings.HasPrefix(key, paxSchilyXattr) {
 			key = key[len(paxSchilyXattr):]
 			if err := setxattr(path, key, value); err != nil {
-				if errors.Is(err, syscall.ENOTSUP) {
+				if errors.Is(err, syscall.ENOTSUP) || errors.Is(err, syscall.EPERM) {
+					// EPERM occurs if modifying xattrs is not allowed, should be ignored.
```

Comment:
Do you know why modifying xattrs is not allowed? I'm not so sure ignoring the error is good.

Reply:
Do you know which xattrs specifically are causing you problems?  Are you having issues specifically with the `security`, `system`, and/or `trusted` xattr namespaces?  These namespaces have separate permission requirements described in the xattr(7) manual page.

Reply:
Failed to set attr on `quay.io/ansible/creator-ee:v0.5.0` image, `/etc/krb5.conf.d/crypto-policies`  operation not permitted.

Reply:
`mody` always ignore EPERM error. https://github.com/moby/moby/blob/master/pkg/archive/archive.go#L782-L790

Reply:
I checked kernel codebase and found that https://github.com/torvalds/linux/blob/de5c208d533a46a074eb46ea17f672cc005a7269/fs/xattr.c#L129. The /etc/krb5.conf.d/crypto-policies is symlink so that kernel return EPERM. 

I think we can align with docket or add the check the file type for that

Reply:
this issue is caused by user. xattr.
According to https://man7.org/linux/man-pages/man7/xattr.7.html, the user. xattr only works with regular file and dir. 

IMO, just ignore EPERM is not right here. We can check it EPERM with different xattr namespaces

Reply:
@fuweid Thanks for the advice, I check both the xattr namespace and the file type.